lib/plugins/helper/open_graph.js

```diff
@@ -119,7 +119,7 @@ function openGraphHelper(options = {}) {
   images = images.map(path => {
     if (!parse(path).host) {
       // resolve `path`'s absolute path relative to current page's url
-      // `path` can be both absolute (starts with `/`) or relative.
+      // `path` can be both absolute (starts with `http://`) or relative (starts with `/`).
       return resolve(url || config.url, path);
```

Comment:
> Should we bring up a resolveURL in hexo-util?

L173 could be replaced with:

``` js
const urlVal = url || config.url;
return urlVal.replace(/\/$/, '') + '/' + path.replace(urlVal, '').replace(/\\/g, '/').replace(/^\//, '')
```

but it only works when both `urlVal` and `path` have the same protocol scheme (e.g. both start with `http://` or both start with `//`).

`url.resolve()` is (surprisingly) robust, I tested:

1. `http://` + `//`
2. `//` + `http://`
3. `http://` + `https://`

---

in this "open_graph.js", `url.resolve()` even robust when `path` variable is an external link, in which it returns external link instead.

---

Other plugins (of this PR) only deal with relative path, not absolute url, so `(a + b).replace(/\\/g, '/').replace(/\/{2,}/g, '/')` works fine. Perhaps can implement `joinPath()`?

---

Also, in other plugins (of this PR), `url.resolve()` is actually not used according to its [purpose](https://nodejs.org/api/url.html#url_url_resolve_from_to), currently it's used as [`path.join()`](https://nodejs.org/api/path.html#path_path_join_paths) plus converting backward to forward slash, whereas this L123 does utilise `url.resolve()` according to its original purpose.